metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -2300,12 +2305,15 @@ class MetalsLanguageServer(
       item <- dependencySources.getItems.asScala
       scalaTarget <- buildTargets.scalaTarget(item.getTarget)
       sourceUri <- Option(item.getSources).toList.flatMap(_.asScala)
+      path = sourceUri.toAbsolutePath
+      _ = buildTargets.addDependencySource(path, item.getTarget)
```

Comment:
It was needed to correctly call `buildTargets.addDependencySource(path, item.getTarget)`.
Otherwise, it registers jar only once for the first build target where it appeared.

Reply:
Ach alright. This should be fine.